src/grids.cpp

```diff
@@ -3119,23 +3134,43 @@ static double read_vgrid_value(PJ_CONTEXT *ctx, const ListOfVGrids &grids,
         }
     }
     if (!grid) {
+        pj_ctx_set_errno(ctx, PJD_ERR_GRID_AREA);
         return HUGE_VAL;
     }
 
     const auto &extent = grid->extentAndRes();
 
-    /* Interpolation a location within the grid */
+    /* Interpolation of a location within the grid */
     double grid_x = (input.lam - extent.westLon) / extent.resLon;
-    if (extent.fullWorldLongitude()) {
-        // The first fmod goes to ]-lim, lim[ range
-        // So we add lim again to be in ]0, 2*lim[ and fmod again
-        grid_x =
-            fmod(fmod(grid_x + grid->width(), grid->width()) + grid->width(),
-                 grid->width());
+    if (input.lam < extent.westLon) {
+        if (extent.fullWorldLongitude()) {
+            // The first fmod goes to ]-lim, lim[ range
+            // So we add lim again to be in ]0, 2*lim[ and fmod again
+            grid_x = fmod(fmod(grid_x + grid->width(), grid->width()) +
+                              grid->width(),
+                          grid->width());
+        } else {
+            grid_x = (input.lam + 2 * M_PI - extent.westLon) / extent.resLon;
+        }
+    } else if (input.lam > extent.eastLon) {
+        if (extent.fullWorldLongitude()) {
+            // The first fmod goes to ]-lim, lim[ range
+            // So we add lim again to be in ]0, 2*lim[ and fmod again
+            grid_x = fmod(fmod(grid_x + grid->width(), grid->width()) +
+                              grid->width(),
+                          grid->width());
+        } else {
+            grid_x = (input.lam - 2 * M_PI - extent.westLon) / extent.resLon;
+        }
     }
     double grid_y = (input.phi - extent.southLat) / extent.resLat;
     int grid_ix = static_cast<int>(lround(floor(grid_x)));
-    assert(grid_ix >= 0 && grid_ix < grid->width());
+    if (!(grid_ix >= 0 && grid_ix < grid->width())) {
+        // Shouldn't happen, but I don't trust asserts
```

Comment:
Why not?

Reply:
> Why not?

Experience :-) Pretty much all assertions become runtime failures if people try to torture the code sufficiently. I suspect here that there could be some floating point issues if trying very close to the grid boundary

Reply:
Fair enough :-) Maybe just remove the comment about not trusting asserts. It makes sense in the diff where the context is clear but when reading this in the future I think it will just be a confusing statement. Instead, "in the unlikely case we end up here, ..."

Reply:
Comment changed as suggested.